File: Frontend/src/@types/celestial/containers/asteroid-belt.ts
import { Asteroid } from "../asteroid";
import { CelestialBody } from "../base/celestial-body";

export interface AsteroidBelt extends Omit<CelestialBody, "axialTilt"> {
  numOfAsteroids: number;
  asteroids?: Asteroid[];
  depth?: number; // in AU
  fill?: boolean;
  minSize?: number; // in KM
  maxSize?: number; // in KM
  inclination?: number; // Degrees
  innerSemiMajor: number; // in AU
  innerEccentricity: number;
  outerSemiMajor: number; // in AU
  outerEccentricity: number;
}

File: Backend/app/models/celestial/asteroid_belt.rb
class Celestial::AsteroidBelt < Celestial::Base::CelestialBody
  self.table_name = 'asteroid_belts'

  [:inner_semi_major, :outer_semi_major, :inner_eccentricity,
   :outer_eccentricity, :inclination, :num_of_asteroids, :depth].each do |attr|
    validates attr, presence: { message: -> (object, _) do
      "The #{object.model_name.human.downcase} must have a value for the #{attr.to_s.humanize.downcase} defined."
    end }, numericality: true
  end
end
